chore(typings): upgrade 'tsd' to 'typings'

NathanWalker opened the pull request:
For the differences between tsd and typings see https://github.com/typings/typings/issues/72

Thanks to @gdi2290 for sharing the 'how to' on this.

NathanWalker committed:
chore(typings): upgrade 'tsd' to 'typings'

robwormald commented on package.json:
```diff
@@ -5,9 +5,11 @@
     "clean_test": "rm -rf tmp/",
     "build_test": "npm run clean_test && tsc -p spec",
     "build_dist": "rm -rf dist && tsc",
+    "postinstall": "typings install && npm run clean-ng2",
```
this seems like a bad idea, as it introduces a dependency on `typings` for consumers of the module, no?

PatrickJS replied:
`typings` is a devDep so it should be fine

NathanWalker replied:
@robwormald I'm also weary about `postinstall` scripts. @gdi2290 If you believe this will not cause any issue, then I trust you :)  In general, I like `typings` 100% over `tsd` and was following @gdi2290 guide on this which I believe to be solid.

PatrickJS commented on src/store.ts:
```diff
@@ -21,7 +21,7 @@ export interface Reducer<T> {
 
 export class Store<T> extends BehaviorSubject<T> {
-	private _storeSubscription: Subscription<T>
+  private _storeSubscription: Subscription;
```
was this fix out of scope of the PR?

NathanWalker replied:
It threw an error with the `<T>`. Wasn't sure why. Removing it made it work.

robwormald replied:
its a change in rx beta 1, iirc.

NathanWalker commented:
Is there anything else needed here for this to merge @robwormald ?

robwormald commented:
still got `postinstall` in there, as far as i can tell. not comfortable forcing the use of `typings` by consumers of the module.

NathanWalker commented:
I removed the `postinstall` step, good call. Instead I've added a `CONTRIBUTING` doc linked from the `README` which outlines those steps in order to setup the repo correctly locally if someone were wanting to contribute.

robwormald commented on package.json:
```diff
@@ -5,6 +5,7 @@
   "main": "./dist/store.js",
   "scripts": {
     "test": "npm run build_test && jasmine",
+    "clean-ng2": "rimraf node_modules/angular2/manual_typings node_modules/angular2/typings",
```
don't see rimraf as a dependency here?

NathanWalker replied:
I'm about to push with `rimraf` included as dep and a cleanup to the `typings.json` to reflect latest changes in typings package names and commits.

NathanWalker replied:
Added as proper dep now, good eye:
https://github.com/NathanWalker/store/commit/57fb4530b73f498500b666998e38a9df9f7db9f9#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R38

robwormald commented on tsconfig.json:
```diff
@@ -9,6 +9,7 @@
 		"declaration": true
 	},
 	"files": [
+    "typings/main.d.ts",
```
is this necessary? afaik the core itself doesn't depend on any external typedefinitions...

NathanWalker replied:
It's needed for `tsc` to compile with no warnings when running `npm run build_dist`.
Otherwise, you get:

```
node_modules/angular2/src/core/change_detection/parser/locals.d.ts(3,14): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/change_detection/parser/locals.d.ts(4,42): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/element_injector.d.ts(71,32): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/element_injector.d.ts(73,17): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/element_injector.d.ts(77,184): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/element_injector.d.ts(82,182): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/element_injector.d.ts(106,37): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/proto_view_factory.d.ts(28,146): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/view.d.ts(52,144): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/view.d.ts(76,79): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/view.d.ts(77,73): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/view.d.ts(95,31): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/view.d.ts(98,18): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/view.d.ts(101,24): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/view.d.ts(104,162): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/linker/view.d.ts(105,160): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/render/api.d.ts(317,74): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/core/zone/ng_zone.d.ts(2,37): error TS2304: Cannot find name 'Zone'.
node_modules/angular2/src/facade/collection.d.ts(1,25): error TS2304: Cannot find name 'MapConstructor'.
node_modules/angular2/src/facade/collection.d.ts(2,25): error TS2304: Cannot find name 'SetConstructor'.
node_modules/angular2/src/facade/collection.d.ts(4,27): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/collection.d.ts(4,39): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/collection.d.ts(7,9): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/collection.d.ts(8,30): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/collection.d.ts(11,43): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/collection.d.ts(12,27): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/collection.d.ts(14,23): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/collection.d.ts(15,25): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/collection.d.ts(94,41): error TS2304: Cannot find name 'Set'.
node_modules/angular2/src/facade/collection.d.ts(95,22): error TS2304: Cannot find name 'Set'.
node_modules/angular2/src/facade/collection.d.ts(96,25): error TS2304: Cannot find name 'Set'.
node_modules/angular2/src/facade/lang.d.ts(2,22): error TS2304: Cannot find name 'BrowserNodeGlobal'.
node_modules/angular2/src/facade/lang.d.ts(48,59): error TS2304: Cannot find name 'Map'.
node_modules/angular2/src/facade/promise.d.ts(1,10): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(3,14): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(8,32): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(9,38): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(10,35): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(10,93): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(11,34): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(12,32): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(12,149): error TS2304: Cannot find name 'Promise'.
node_modules/angular2/src/facade/promise.d.ts(13,43): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/CoreOperators.d.ts(22,67): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/CoreOperators.d.ts(72,67): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/CoreOperators.d.ts(77,31): error TS2304: Cannot find name 'PromiseConstructor'.
node_modules/rxjs/CoreOperators.d.ts(77,54): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(65,67): error TS2304: Cannot find name 'PromiseConstructor'.
node_modules/rxjs/Observable.d.ts(65,88): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(72,84): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(77,38): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(100,66): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(154,66): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(159,31): error TS2304: Cannot find name 'PromiseConstructor'.
node_modules/rxjs/Observable.d.ts(159,54): error TS2304: Cannot find name 'Promise'.
src/store.ts(42,25): error TS2339: Property 'assign' does not exist on type 'ObjectConstructor'.
src/store.ts(50,29): error TS2339: Property 'assign' does not exist on type 'ObjectConstructor'.
```

NathanWalker commented:
PR now contains the most up-to-date `typings` config. Verified the setup with a clean slate and tests pass, as well as the build completes.

NathanWalker commented on spec/tsconfig.json:
```diff
@@ -8,8 +8,9 @@
 		"outDir": "../tmp"
 	},
 	"files": [
+    "../typings/main.d.ts",
```
This is also needed for the tests to compile and run.

robwormald commented:
Cool. Beta4 brings in some changes in typedefs, so I reckon we merge and release this to sync up with that. See https://goo.gl/70dIcF - might want to check against that

robwormald commented:
@NathanWalker / @gdi2290 can you explain why you're using a standalone version of "ng2-typings"? what is this precisely?

PatrickJS commented:
because we're opting out of the typings included in the angular2 module `facade/lang` still requires a `BrowserNodeGlobal` type
![screen shot 2016-02-07 at 7 46 59 pm](https://cloud.githubusercontent.com/assets/1016365/12878109/9824d0f8-cdd3-11e5-8cc4-5b0dde9545de.jpg)
